articles/understanding-systemd/index.md

```diff
@@ -0,0 +1,291 @@
+
+There are default unit files supplied by the packages of the service that usually reside in **/lib/systemd/system/**.  These files are vendor-written and can be overwritten by the package maintainer in the event of a package update. User-specific configurations for units are usually made in **/etc/systemd/system/** because the unit files in this location won't be overwritten in the event of a package upgrade.
+
+**Note:** It's common practice to copy the default unit file into **/etc/systemd/system/** and make appropriate customization.
+
+There are 11 type of units in systemd. We'd talk about the most common ones. ***man 1 systemd*** describes them. So give it a check.
+
+1. Service units: This unit defines how to manage service daemons controlled by systemd. Unit files of this service end with .service.
+2. Socket units: This contains information about a local inter-process communication or network sockets in a Linux system.
+3. Target units: This is used to provide synchronization amongst unit files during boot-up. It can also be used as a medium to extend scope by specifying their targets to other targets.
+4. Device units: This unit describes devices that need exposure to systemd by udev and sysfs filesystem.
+5. Mount units: This defines a mount point for attaching filesystems controlled by systemd.
+6. Automount units: This provides automount capabilities.
+7. Timer units: This unit (synonymous with cron) defines a timer controlled by systemd. It schedules the activation of other units.
+8. Swap units: This unit describes a memory swap partition.
+9. Path units: This can define a path that activates other services when files within the path are modified. 
+10. Slice units: Now this is a special unit. It's used in conjunction with cgroups to group processes, daemons, services into one hierarchical tree for the management of resources.
+11. Scope units: This manages externally created processes. See man 5 systemd.scope.
+
+---
+
+### How systemd manages dependencies
+
+Earlier on, we talked about systemd starting up services required to hit a default boot goal during boot time. Imagine this default boot goal to be a node at the top of a tree. So when a target is activated, all other targets below it (dependencies) are also activated.
+
+A visual cue could be of help here.
+
+![Figure 1. Dependency tree](/engineering-education/understanding-systemd/)
+
+So it turns out that dependency management is difficult in modern Linux machines. These machines have many services that depend on each other. One service failure can solely freeze the boot process of a Linux machine.
+
+Imagine defining a dependency from an X server to a logging service. That is, the logging service would have to start before the X server. In a situation where the logging service doesn't start, we'd be unable to get our X server running because one of its dependencies isn't running. In this situation, X server is a target for the logging service.
```

Comment:
You're missing an article here '.........X server is a target for the logging service.', X server is like a noun phrase and shouldn't miss a determiner before it.